From 35ca0d38bd0779af0555e1036254719062b0e2c1 Mon Sep 17 00:00:00 2001
From: Akatsuki <3736910+akiirui@users.noreply.github.com>
Date: Thu, 5 Dec 2019 18:02:35 +0800
Subject: [PATCH] 2

---
 gtk/torrent-cell-renderer.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/gtk/torrent-cell-renderer.c b/gtk/torrent-cell-renderer.c
index f5b55b2..01f4ea1 100644
--- a/gtk/torrent-cell-renderer.c
+++ b/gtk/torrent-cell-renderer.c
@@ -420,7 +420,7 @@ get_size_compact (TorrentCellRenderer * cell,
 
 #define BAR_WIDTH 50
     if (width != NULL)
-        *width = xpad * 2 + icon_size.width + GUI_PAD + name_size.width + GUI_PAD + BAR_WIDTH + GUI_PAD + stat_size.width;
+        *width = xpad * 2 + icon_size.width + GUI_PAD + BAR_WIDTH + GUI_PAD + stat_size.width;
     if (height != NULL)
         *height = ypad * 2 + MAX (name_size.height, p->bar_height);
 
@@ -428,8 +428,6 @@ get_size_compact (TorrentCellRenderer * cell,
     g_object_unref (icon);
 }
 
-#define MAX3(a,b,c) MAX(a,MAX(b,c))
-
 static void
 get_size_full (TorrentCellRenderer * cell,
                GtkWidget           * widget,
@@ -474,7 +472,7 @@ get_size_full (TorrentCellRenderer * cell,
     **/
 
     if (width != NULL)
-        *width = xpad * 2 + icon_size.width + GUI_PAD + MAX3 (name_size.width, prog_size.width, stat_size.width);
+        *width = xpad * 2 + icon_size.width + GUI_PAD + MAX (prog_size.width, stat_size.width);
     if (height != NULL)
         *height = ypad * 2 + name_size.height + prog_size.height + GUI_PAD_SMALL + p->bar_height + GUI_PAD_SMALL + stat_size.height;
 
@@ -925,4 +923,3 @@ torrent_cell_renderer_new (void)
     return (GtkCellRenderer *) g_object_new (TORRENT_CELL_RENDERER_TYPE,
                                              NULL);
 }
-
-- 
2.24.0

