From 59804cb0581076f93dfded6af18313596e9a0921 Mon Sep 17 00:00:00 2001
From: Akatsuki <3736910+akiirui@users.noreply.github.com>
Date: Sat, 7 Dec 2019 22:30:10 +0800
Subject: [PATCH] center-fix

---
 src/core/keybindings.c | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/src/core/keybindings.c b/src/core/keybindings.c
index b86272541..00c2fc857 100644
--- a/src/core/keybindings.c
+++ b/src/core/keybindings.c
@@ -3085,16 +3085,25 @@ handle_move_to_center  (MetaDisplay     *display,
                         MetaKeyBinding  *binding,
                         gpointer         dummy)
 {
-  MetaRectangle work_area;
+  int w, h, x, y;
+  MetaBackend *backend = meta_get_backend ();
+  MetaLogicalMonitor *logical_monitor;
   MetaRectangle frame_rect;
 
-  meta_window_get_work_area_all_monitors (window, &work_area);
   meta_window_get_frame_rect (window, &frame_rect);
 
+  /* Warning, this function is a round trip! */
+  logical_monitor = meta_backend_get_current_logical_monitor (backend);
+
+  w = logical_monitor->rect.width;
+  h = logical_monitor->rect.height;
+  x = logical_monitor->rect.x;
+  y = logical_monitor->rect.y;
+
   meta_window_move_frame (window,
                           TRUE,
-                          work_area.x + (work_area.width  - frame_rect.width ) / 2,
-                          work_area.y + (work_area.height - frame_rect.height) / 2);
+                          x + (w - frame_rect.width ) / 2,
+                          y + (h - frame_rect.height) / 2);
 }
 
 static void
-- 
2.24.0

